ee/agent/knapsack/knapsack.go

```diff
@@ -262,10 +262,10 @@ func (k *knapsack) SetEnrollmentDetails(details types.EnrollmentDetails) {
 
 	k.Slogger().Log(context.Background(), slog.LevelDebug,
 		"updating enrollment details",
-		"old_details", fmt.Sprintf("%+v", enrollmentDetails),
-		"new_details", fmt.Sprintf("%+v", current),
+		"old_details", fmt.Sprintf("%+v", current),
```

Comment:
nit- I think this entire function might be more readable/less error prone as something like 
```
func (k *knapsack) SetEnrollmentDetails(newDetails types.EnrollmentDetails) {
	k.Slogger().Log(context.Background(), slog.LevelDebug,
		"updating enrollment details",
		"old_details", fmt.Sprintf("%+v", enrollmentDetails),
		"new_details", fmt.Sprintf("%+v", newDetails),
	)

	enrollmentDetails = &newDetails
}
```

Reply:
i also don't mind keeping the separate initializing log if that's more clear to people than old_details being nil, but it felt like most of the pointer swapping was not needed

Reply:
Sure thing -- updated

Reply:
It felt clear enough to me in the log without the separate initializing log --

```
"msg":"updating enrollment details","old_details":"<nil>","new_details":"{OSVersion: OSBuildID: OSPlatform:darwin Hostname: HardwareVendor: HardwareModel: HardwareSerial: OsqueryVersion: LauncherHardwareKey: LauncherHardwareKeySource: LauncherLocalKey: LauncherVersion:1.15.2-31-g25c6278d-dirty OSName: OSPlatformLike:darwin GOOS:darwin GOARCH:arm64 HardwareUUID:}"
```